doc/vfd-swmr-user-guide.md

```diff
@@ -0,0 +1,496 @@
+use case, the HDF5 file is opened in VFD SWMR reader mode by a second program 
+that generates a real time display of the data as it is being collected -- thus 
+allowing the experimenters to steer the experiment.
+
+THG developed a demonstration program for class of application,
+and we have some advice based on that experience. 
+
+The writer typically will increase a dataset's dimensions by a
+frame, using `H5Dset_extent()`, before it writes the data of that
+frame with `H5Dwrite()`.  It's possible that a snapshot of the HDF5
+file will propagate to the reader between the `H5Dset_extent()`
+call and the `H5Dwrite()`.  Values `H5Dread()` from the last frame
+at that juncture will not reflect the actual experimental data.
+Instead, the reader will see arbitrary values or the fill value.
+To display those values would be distracting and misleading to
+the experimenter. 
+
+On the reader, a strategy for displaying the most current, bonafide application
+data is to read the dimensions of the frames dataset, `d`, compute
+the number `n` of full frames contained in `d`, and read the
+next-to-last frame, `n - 2`.  THG uses a variant of this strategy
+in its `gaussians` demo.
+
+On the writer, a strategy for protecting against snapshots between
+the `H5Dset_extent()` and `H5Dwrite()` calls is to suspend VFD
+SWMR's clock across both of the calls.  The
+`H5Fvfd_swmr_disable_end_of_tick()` call takes a file identifier
+and stops new snapshots from being taken on the given file until
+`H5Fvfd_swmr_enable_end_of_tick()` is called on the same file.
+Needless to say, end of tick processing should only be disabled
+briefly.
+
+# Known issues
+
+## Variable-length data
+
+A VFD SWMR reader cannot reliably read back a variable-length dataset
+written by VFD SWMR.  For example, a variable-length string
+created and written as follows
+
+```
+    hid_t dset, space, type;
+    char data[] = "content";
+
+    type = H5Tcopy(H5T_C_S1);
+
+    H5Tset_size(type, H5T_VARIABLE);
+
+    space = H5Screate(H5S_SCALAR);
+
+    dset = H5Dcreate2(..., "string", type, space, H5P_DEFAULT, H5P_DEFAULT,
+        H5P_DEFAULT);
+
+    H5Dwrite(dset, type, space, space, H5P_DEFAULT, &data);
+```
+
+and read back like this,
+
+```
+    char *data;
+    herr_t ret;
+
+    ret = H5Dread(..., ..., H5S_ALL, H5S_ALL, H5P_DEFAULT, &data);
+```
+
+may produce either an error return from `H5Dread` (`ret < 0`) or
+a `NULL` pointer (`data == NULL`).
+
+As discussed above, this is caused by a fundamental incompatibility 
+between the current variable length data implementation in HDF5, which 
+stores variable length data as metadata.  It is possible we may be able 
+to mitigate the issue, but the most likely solution is the planned 
+re-implementation of variable length data that is currently in the planning
+stage.  Unfortunately, we have no ETA for this re-implementation.
+
+## Iteration
+
+An application that reads in VFD SWMR mode should take care to avoid
```

Comment:
Then just have fail when the file is opened with the SWMR VFD.   Consistent behavior is better than telling users "random stuff might haapen".